router/http/src/main/scala/io/buoyant/router/http/ViaHeaderAppenderFilter.scala

```diff
@@ -39,8 +38,7 @@ object ViaHeaderAppenderFilter {
 
     def apply(req: Request, svc: Service[Request, Response]) = {
       ViaLinkerd.appendViaHeader(req)
-      req.version = Http11
-      if (req.host.isEmpty) {
+      if (req.version == Http11 && req.host.isEmpty) {
```

Comment:
Let's add a comment here explaining that we forward HTTP/1.0 requests using HTTP/1.0 to minimize the chance that the server will send a response that uses HTTP/1.1 features, in particular `Transfer-Encoding: chunked`, that the HTTP/1.0 application can't handle.